sdk/opa.go

```diff
@@ -277,6 +286,9 @@ type DecisionOptions struct {
 	NDBCache            interface{}         // specifies the non-deterministic builtins cache to use for evaluation.
 	StrictBuiltinErrors bool                // treat built-in function errors as fatal
 	Tracer              topdown.QueryTracer // specifies the tracer to use for evaluation, optional
+	Metrics             metrics.Metrics     // specifies the metrics to use for preparing and evaluation, optional
+	Profiler            *profiler.Profiler  // specifies the profiler to use, optional
```

Comment:
I opted to expose the Profiler as a `profiler.Profiler` rather than a `topdown.QueryTracer` to make it clear how this is meant to be used. However, internally, it's passed as a `topdown.QueryTracer` to keep the implementation generic.

Reply:
I think it might be worthwhile to consider using more interfaces here. After all, a user of the SDK could provide their own implementation, perhaps...? 🤔

Reply:
I think b0aade94f32d7937bcb4319618f00b864e67db29 looks better ✅